### PR TITLE
chore(test): Run a single E2E test to verify the AoT build loads

### LINKED ISSUE #1212: CI: Run a single E2E test to verify the AoT build loads
We have had two AoT bugs slip through our CI process. This is because those specific items weren't errors at build time, but were errors when the app was actually loaded. We should use an E2E framework to boot up the app and verify it doesn't error.

- [ ] Add single E2E test to the CI flow
  - eg Load page and check for text.

### PR DESCRIPTION
**Pull request by atlwendy**
closes #1212

### CODE REVIEW

File: .github/workflows/ci-push.yml [hunk 1]
```diff
@@ -192,3 +192,51 @@ jobs:
...
+        id: cache_node_modules
+        uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node-${{ hashFiles('**/yarn.lock') }}
+        if: env.SHOULD_RUN == 'true'
+      - if: env.SHOULD_RUN == 'true' && steps.cache_node_modules.outputs.cache-hit != 'true'
+        run: yarn install --frozen-lockfile --non-interactive
+      - if: env.SHOULD_RUN == 'true'
+        run: yarn add start-server-and-test
```
Comment (benjamincharity) [inline .github/workflows/ci-push.yml:227]: Any reason we are installing here and not adding as a devDependency?

File: lint-staged.config.js [hunk 2]
```diff
@@ -46,16 +47,14 @@ module.exports = {
...
  */
-function removeDirectories(files) {
-  const match = micromatch.not(files, [
-    ...IGNORE,
-    '**/testing/**',
-    '**/demo/**',
-  ]);
-  return match.join(' ');
-}
+const removeDirectories = files => micromatch.not(files, [
```
Comment (benjamincharity) [inline lint-staged.config.js:55]: Why were changes needed here?

**Review by benjamincharity — CHANGES_REQUESTED** [review]
Couple small questions inline.

One larger question here: Do you think we should wait on using Cypress for the release flow until we have a better handle on it? I'm worried that it will kill our automated releases and I'll have to sort through partially released versions. Thoughts?

File: .github/workflows/ci-push.yml (hunk 1 above)
Comment (atlwendy) [reply .github/workflows/ci-push.yml:227]: I thought about that, but this is more just for starting server on ci... no other place it'd be needed. I'm okay either way if you prefer one over the other...

File: lint-staged.config.js (hunk 2 above)
Comment (atlwendy) [reply lint-staged.config.js:55]: It was returning string, but I'd want it to be array, so can directly plug into the function you defined where it targets all ts and js files. And it doesn't seem it was referenced anywhere else?

**Review by benjamincharity — APPROVED** [review]

**Pull request merged**